compiler-lib/src/type_checking/type_system.rs

```diff
@@ -168,7 +183,8 @@ pub enum CheckedType<'src> {
     Boolean,
     Void,
     Null,
-    TypeRef(Symbol<'src>),
+    TypeRef(ClassDefId<'src>),
+    UnknownType(Symbol<'src>),
```

Comment:
pleas add a comment *right here*  what this is, and whether we need to take care of it in the ir codegen.
if not, please change the public API such that this implementation detail is not leaked.

otherwise, to get exhaustive enum matches, we need to panic wherever we match on `CheckedType`

comments?

Reply:
`UnknownType` is only constructed in `checked_type_from_ty` if `lookup_class` fails. This shouldn't happen at all IIUC.

Reply:
See #143